src/middlewared/middlewared/plugins/vmware.py

```diff
@@ -194,7 +191,14 @@ def get_datastores(self, data):
             datastores[esxi_host.name] = datastores_host
 
         connect.Disconnect(server_instance)
-        return datastores
+
+        # Datastore names are unique among different esxi host under a single vcenter server. As we only require
+        # datastore names and additional information is not needed for datastore choices, we refine the datastores
+        # dict and send back only the datastore names
+
+        return sorted({
+            name: name for host in datastores.values() for name in host
```

Comment:
whats the point in making it a dict if you are then turning it into an array?

Reply:
Removing duplicates

Reply:
you use a set for that, not a dict